File: packages/react-native-nitro-test/android/src/main/java/com/margelo/nitro/test/HybridExternalChild.kt
package com.margelo.nitro.test

import androidx.annotation.Keep
import com.facebook.proguard.annotations.DoNotStrip

@Keep
@DoNotStrip
class HybridExternalChild : HybridExternalChildSpec() {
  override val baseValue: Double
    get() = 40.0

  override fun bounceString(string: String): String {
    return string
  }

  override fun toString(): String {
    return "HybridExternalChild custom toString() :)"
  }
}

File: packages/react-native-nitro-test/ios/HybridExternalChild.swift
//
//  HybridExternalChild.swift
//  react-native-nitro-test
//
//  Created by Marc Rousavy on 03.10.24.
//

import Foundation

class HybridExternalChild: HybridExternalChildSpec {
  var baseValue: Double {
    return 40
  }

  func bounceString(string: String) throws -> String {
    return string
  }

  func toString() -> String {
    return "HybridExternalChild custom toString() :)"
  }
}
